block-streamer/src/receive.rs

```diff
@@ -15,7 +17,13 @@ pub async fn receive(listen_addr: &String) -> Result<()> {
 
     let mut buf = vec![0; MTU];
     let mut real_len;
-    let mut data_receiver = Receiver::new();
+
+    // Setup storage
+    let provider = SqliteStorageProvider::new("storage.db")?;
```

Comment:
There is some duplication with `control.rs`, consider a reusable abstraction at some point.

Reply:
Yup I totally agree...I think its likely that this file will go away and `control` will be the default path for handling data protocol messages. It already does that alongside the API messages.